### PR TITLE
Add date time form field

### PR DESCRIPTION
**Pull request by Devessier**
- Create a really simple abstraction to unify the date and date time fields. We might dissociate them sooner than expected.
- The _relative_ setting is ignored

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormDateishFieldInputBase.tsx [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+type FormDateishFieldInputBaseProps = {
+  mode: 'date' | 'datetime';
+  label?: string;
+  placeholder?: string;
+  defaultValue: string | undefined;
+  onPersist: (value: string | null) => void;
+  VariablePicker?: VariablePickerComponent;
+};
+
+export const FormDateishFieldInputBase = ({
```
Comment (martmull) [inline packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormDateishFieldInputBase.tsx:81]: This naming is not very clear. See my global comment
```suggestion
export const FormDateTimeFieldInput = ({
```

**Review by martmull — CHANGES_REQUESTED** [review]
Hey nice code. I think the components logic can be simplified. Delete `FormDateFieldInput.tsx` and `FormDateTimeFieldInput.tsx` then rename `FormDateishFieldInputBase.tsx` into `FormDateTimeFieldInput.tsx`
In `FormFieldInput.tsx` use `<FormDateTimeFieldInput mode='datetime' ...>` and `<FormDateTimeFieldInput mode='date' ...>`

**Comment by thomtrp** [comment]
> Hey nice code. I think the components logic can be simplified. Delete `FormDateFieldInput.tsx` and `FormDateTimeFieldInput.tsx` then rename `FormDateishFieldInputBase.tsx` into `FormDateTimeFieldInput.tsx` In `FormFieldInput.tsx` use `<FormDateTimeFieldInput mode='datetime' ...>` and `<FormDateTimeFieldInput mode='date' ...>`

+1 on this

**Review by martmull — COMMENTED** [review]
LGTM thanks

**Review by martmull — APPROVED** [review]

**Pull request merged**